packages/mermaid/src/diagrams/flowchart/flowDb.ts

```diff
@@ -16,1101 +18,1082 @@ import {
-let classes = new Map<string, FlowClass>();
-let subGraphs: FlowSubGraph[] = [];
-let subGraphLookup = new Map<string, FlowSubGraph>();
-let tooltips = new Map<string, string>();
-let subCount = 0;
-let firstGraphFlag = true;
-let direction: string;
-
-let version: string; // As in graph
-
-// Functions to be run after graph rendering
-let funs: ((element: Element) => void)[] = []; // cspell:ignore funs
-
-const sanitizeText = (txt: string) => common.sanitizeText(txt, config);
-
-/**
- * Function to lookup domId from id in the graph definition.
- *
- * @param id - id of the node
- */
-export const lookUpDomId = function (id: string) {
-  for (const vertex of vertices.values()) {
-    if (vertex.id === id) {
-      return vertex.domId;
-    }
-  }
-  return id;
-};
-
-/**
- * Function called by parser when a node definition has been found
- */
-export const addVertex = function (
-  id: string,
-  textObj: FlowText,
-  type: FlowVertexTypeParam,
-  style: string[],
-  classes: string[],
-  dir: string,
-  props = {},
-  metadata: any
-) {
-  if (!id || id.trim().length === 0) {
-    return;
-  }
-  // Extract the metadata from the shapeData, the syntax for adding metadata for nodes and edges is the same
-  // so at this point we don't know if it's a node or an edge, but we can still extract the metadata
-  let doc;
-  if (metadata !== undefined) {
-    let yamlData;
-    // detect if shapeData contains a newline character
-    if (!metadata.includes('\n')) {
-      yamlData = '{\n' + metadata + '\n}';
-    } else {
-      yamlData = metadata + '\n';
-    }
-    doc = yaml.load(yamlData, { schema: yaml.JSON_SCHEMA }) as NodeMetaData;
-  }
 
-  // Check if this is an edge
-  const edge = edges.find((e) => e.id === id);
-  if (edge) {
-    const edgeDoc = doc as EdgeMetaData;
-    if (edgeDoc?.animate) {
-      edge.animate = edgeDoc.animate;
-    }
-    if (edgeDoc?.animation) {
-      edge.animation = edgeDoc.animation;
-    }
-    return;
+// We are using arrow functions assigned to class instance fields instead of methods as they are required by flow JISON
+export class FlowDb implements DiagramDB {
+  private vertexCounter = 0;
+  private config = getConfig();
+  private vertices = new Map<string, FlowVertex>();
+  private edges: FlowEdge[] & { defaultInterpolate?: string; defaultStyle?: string[] } = [];
+  private classes = new Map<string, FlowClass>();
+  private subGraphs: FlowSubGraph[] = [];
+  private subGraphLookup = new Map<string, FlowSubGraph>();
+  private tooltips = new Map<string, string>();
+  private subCount = 0;
+  private firstGraphFlag = true;
+  private direction: string | undefined;
+
+  private version: string | undefined; // As in graph
+
+  private secCount = -1;
+  private posCrossRef: number[] = [];
+
+  // Functions to be run after graph rendering
+  private funs: ((element: Element) => void)[] = []; // cspell:ignore funs
+
+  constructor() {
+    this.funs.push(this.setupToolTips);
+    this.clear();
```

Comment:
getConfig is called when initailizing, and also inside this.clear function, so one call is redundant.